### PR TITLE
Schwim Fixes

### PR DESCRIPTION
**Pull request by shweinfeld**

### CODE REVIEW

File: src/main/java/pong/Ball.java [hunk 1]
```diff
@@ -59,4 +55,27 @@ public void move() {
...
+                        this.getY() <= opponent.getY() + PongView.PADDLE_HEIGHT)) {
+
+            this.toggleVelX();
+        }
+    }
+    public boolean checkBallOutOfBounds(Paddle user, Paddle opponent) { //aka did someone score a point
+        return this.getX() > user.getX() + this.MID_OF_BALL || this.getX() < opponent.getX() + this.MID_OF_BALL;
+    }
+
+    public void checkWellCollide(){
```
Comment (dnagelCS) [inline src/main/java/pong/Ball.java]: this seems like a mistake. should be checkWallCollide()

File: src/main/java/pong/Environment.java [hunk 2]
```diff
@@ -11,21 +11,28 @@
...
     private int compScore;
+    private boolean scored = false;
 
     public Environment(Paddle user, Paddle opponent, Ball ball) {
         this.user = user;
         this.opponent = opponent;
         this.ball = ball;
         userScore = 0;
         compScore = 0;
+
```
Comment (dnagelCS) [inline src/main/java/pong/Environment.java]: remove whitespace

File: src/main/java/pong/Environment.java [hunk 3]
```diff
@@ -11,21 +11,28 @@
...
     }
 
     public int getUserScore() {
         return userScore;
     }
 
+    public boolean hasScored() {
+        return scored;
+    }
+
```
Comment (dnagelCS) [inline src/main/java/pong/Environment.java:31]: remove whitespace

File: src/main/java/pong/Environment.java [hunk 4]
```diff
@@ -51,11 +59,13 @@ public Paddle getOpponent() {
...
         if (moveBall()) {
             user.move();
-            opponent.move();
             opponent.getStrategy().directPaddle(opponent, ball);
+            opponent.move();
+            scored = false;
         } else { //point has been scored
             if (ball.getX() > user.getX()) {
                 compScored();
+
```
Comment (dnagelCS) [inline src/main/java/pong/Environment.java]: remove whitespace

File: src/main/java/pong/OpponentStrategy.java [hunk 5]
```diff
@@ -1,46 +1,19 @@
...
-            }
-            x += velX;
-            y += velY;
-        }
-        
-        return (int) y;
+        } else {
+             paddle.setUpAccel(false);
+             paddle.setDownAccel(false);
+         }
```
Comment (rachel613g) [inline src/main/java/pong/OpponentStrategy.java:17]: Ah, this strategy is so simple, yet so sweet. Just be where the ball is at!

**Review by rachel613g — COMMENTED** [review]
I like how you guys removed the Player enum, it was so not necessary anymore. Awesome edits, thanks!

**Review by lliebman — APPROVED** [review]

**Pull request merged**